service/Microsoft.Azure.Devices/AmqpServiceClient.cs

```diff
@@ -101,8 +101,10 @@ public SendingAmqpLink SendingLink
 
         public async override Task CloseAsync()
         {
+            this.faultTolerantSendingLink.Dispose();
```

Comment:
>this.faultTolerantSendingLink.Dispose(); [](start = 12, length = 40)

Disposing the sending link in not graceful. Also, it is not needed. Closing the Amqp connection will close all links and sessions inside it.